resources/windows/wix/registryEntries.wxs.erb

```diff
@@ -0,0 +1,99 @@
+<?xml version='1.0' encoding='windows-1252'?>
+<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi' xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+  <!--
+    Puppet Agent Specific Registry Entries
+    This overrides the generic file in the vanagon directory
+  -->
+  <Fragment>
+    <ComponentGroup Id="RegistryComponentGroup">
+      <Component Id="RegistryEntries" Guid="68F5D91C-49C8-43F7-940A-481007689E79" Win64="no">
+        <RegistryKey
+          Root="HKLM"
+          Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:product_name] %>"
+          Action="create" >
+          <!-- This is the default (aka 'unnamed') key value of this path -->
+          <RegistryValue
+            Type="integer"
+            Value="0"/>
+          <!-- This is the specified value of a key at this path -->
+          <RegistryValue
+            Name="InstalledFlag"
+            Value="1"
+            Type="integer"
+            KeyPath="yes"/>
+          <!-- Store properties for later recall during uninstall, upgrade, repair -->
+          <RegistryValue
+            Name="RememberedPuppetMasterServer"
+            Type="string"
+            Value="[PUPPET_MASTER_SERVER]" />
+          <RegistryValue
+            Name="RememberedPuppetAgentEnvironment"
+            Type="string"
+            Value="[PUPPET_AGENT_ENVIRONMENT]" />
+          <RegistryValue
+            Name="RememberedPuppetCaServer"
+            Type="string"
+            Value="[PUPPET_CA_SERVER]" />
+          <RegistryValue
+            Name="RememberedPuppetAgentCertname"
+            Type="string"
+            Value="[PUPPET_AGENT_CERTNAME]" />
+          <RegistryValue
+            Name="RememberedPuppetAgentStartupMode"
+            Type="string"
+            Value="[PUPPET_AGENT_STARTUP_MODE]" />
+        </RegistryKey>
+
+        <RegistryKey
+            Root="HKLM"
+            Key="SYSTEM\CurrentControlSet\services\eventlog\Application\Puppet"
+            Action="createAndRemoveOnUninstall">
+          <RegistryValue
+            Type="string"
+            Name="EventMessageFile"
+            Value="[INSTALLDIR]misc\puppetres.dll"/>
+          <RegistryValue
+            Type="integer"
+            Name="TypesSupported"
+            Value="7"/>
+        </RegistryKey>
+      </Component>
+
+      <Component
+        Id="RegistryEntriesArchitectureDependent"
+        Directory="TARGETDIR"
+        Guid="E6D5AF4F-ACC4-4D11-AFCE-299A9ED2152C"
+        Win64="<%= settings[:win64] %>"
+        Permanent="yes">
+
+        <RegistryKey
```

Comment:
Don't follow you @McdonaldSeanp about line 62, but it is deliberate - remember - this file will override whats in Vanagon, so anything in there (as an example) will have to be repeated here.

Hold off on merge of this - its not quite compiling yet, so have put a Do Not Merge on it.